timeline_server.py

```
@@ -1,14 +1,116 @@
+    escape hatch for deliberately running a deployment open."""
+    if os.getenv("TIMELINE_API_TOKEN", "").strip():
+        return
+
+    if _is_deployment() and os.getenv("TIMELINE_ALLOW_INSECURE", "").strip() != "1":
+        raise RuntimeError(
+            "TIMELINE_API_TOKEN is not set but this looks like a deployment "
+            f"({', '.join(m for m in _DEPLOY_MARKERS if os.getenv(m, '').strip())}). "
+            "The /v1 approval endpoints authorize agent actions and would be "
+            "reachable anonymously. Set TIMELINE_API_TOKEN, or set "
+            "TIMELINE_ALLOW_INSECURE=1 to override deliberately."
+        )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)